Add page height & full width options to blocks

PeeterPaal opened the pull request:
Allow additional settings for blocks. All blocks can be set either full height or "auto" height. Columns block can also be se to either full width or fixed width.

<img width="1570" alt="Screenshot 2025-01-29 at 09 09 15" src="https://github.com/user-attachments/assets/117abee5-0c16-4ec7-9924-f847706b23ea" />


Closes #13

PeeterPaal committed:
Add page height & full width options to blocks

PeeterPaal committed:
Make new block settings buttons functional

tanelj commented on components/block.tpl:
```diff
@@ -1,6 +1,18 @@
 {%- assign id = block_data.key -%}
 {%- assign content_area_count = layout_data.content_areas | default: 1 -%}
 
+{% assign page_height = block_data.settings.page_height %}
```
```suggestion
{%- assign page_height = block_data.settings.page_height -%}
```

tanelj commented on components/block.tpl:
```diff
@@ -1,6 +1,18 @@
 
+{% assign page_height = block_data.settings.page_height %}
+
+{% if page_height == null %}
```
```suggestion
{%- if page_height == null %}
```

tanelj commented on components/block.tpl:
```diff
@@ -1,6 +1,18 @@
+
+{% if page_height == null %}
+  {% assign page_height = true %}
+{% endif %}
```
```suggestion
{% endif -%}
```

tanelj commented on components/block.tpl:
```diff
@@ -1,6 +1,18 @@
+  {% assign page_height = true %}
+{% endif %}
+
+{% assign full_width = block_data.settings.full_width %}
```
```suggestion
{%- assign full_width = block_data.settings.full_width -%}
```

tanelj commented on components/block.tpl:
```diff
@@ -1,6 +1,18 @@
+
+{% assign full_width = block_data.settings.full_width %}
+
+{% if full_width == null %}
```
```suggestion
{%- if full_width == null %}
```

tanelj commented on components/block.tpl:
```diff
@@ -1,6 +1,18 @@
+
+{% if full_width == null %}
+  {% assign full_width = false %}
+{% endif %}
```
```suggestion
{% endif -%}
```

tanelj reviewed (changes requested):
Nice and works. Only some micro cleanup.

PeeterPaal committed:
Improve accessing layout settings

Firefox does not hande select menus on hover well. Add `active` class
to block in editmode to not solely rely on hover.

PeeterPaal committed:
Reduce empty lines in output HTML

PeeterPaal committed:
Use opacity for edit buttons

tanelj reviewed (approved):
Nice!
